### PR TITLE
Update 3rdparty libraries

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Forked the original github repo's and added them as submodules.

This allows for easier updating of these 3rdparty libraries in the future.

### CODE REVIEW

File: src/io/json_reader.cpp [hunk 1]
```diff
@@ -34,6 +32,11 @@ struct MyHandler {
 
     bool Double(double d) { w.setValue(key, d); return true; }
 
+    bool RawNumber(const char* str, rapidjson::SizeType len, bool copy)
+    {
+        w.setValue(key, str);
+        return true;
+    }
```
Comment (PetervDooren) [inline src/io/json_reader.cpp:39]: What is this for? It seems unrelated to the 3rd party changes

File: include/ed/io/json_reader.h [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace ed
 namespace io
 {
 
-class JSONReader : public Reader
+class JSONReader : public ed::io::Reader
```
Comment (PetervDooren) [inline include/ed/io/json_reader.h:16]: does this make any difference given the namespaces here? And if not, why leave the namespaces here?

File: src/io/json_reader.cpp [hunk 3]
```diff
@@ -91,7 +94,7 @@ struct MyHandler {
         return true;
     }
 
-    DataWriter& w;
+    ed::io::DataWriter& w;
```
Comment (PetervDooren) [inline src/io/json_reader.cpp:97]: same question with namespaces.

File: src/io/json_reader.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply src/io/json_reader.cpp:39]: It is, see commit message of https://github.com/tue-robotics/ed/pull/144/commits/2a4bdb4149009bbca3e68ab5a1df720f498bd640

File: include/ed/io/json_reader.h (hunk 2 above)
Comment (MatthijsBurgh) [reply include/ed/io/json_reader.h:16]: No, it doesn't. But as there is also a Reader class in the rapidjson library(https://github.com/tue-robotics/rapidjson/blob/master/include/rapidjson/reader.h#L2233), it better to make it explicit.

File: src/io/json_reader.cpp (hunk 3 above)
Comment (MatthijsBurgh) [reply src/io/json_reader.cpp:97]: This isn't needed. Though I think it is more clear to use the full NS.

**Review by PetervDooren — APPROVED** [review]
